tests/CSSList/DocumentTest.php

```diff
@@ -85,4 +86,72 @@ public function setContentsReplacesContentsSetInPreviousCall()
 
         self::assertSame($contents2, $this->subject->getContents());
     }
+
+    /**
+     * @return Generator
+     */
+    public static function insertDataProvider(): Generator
+    {
+
+        $bogusOne = new DeclarationBlock();
+        $bogusOne->setSelectors('.bogus-one');
+        $bogusTwo = new DeclarationBlock();
+        $bogusTwo->setSelectors('.bogus-two');
+
+        $oItem = new DeclarationBlock();
+        $oItem->setSelectors('.item');
+
+        $oSibling = new DeclarationBlock();
+        $oSibling->setSelectors('.sibling');
+
+        $oOrphan = new DeclarationBlock();
+        $oOrphan->setSelectors('.forever-alone');
+
+        yield 'insert before' => [
+            'initialContent' => [$bogusOne, $oSibling, $bogusTwo],
+            'oItem' => $oItem,
+            'oSibling' => $oSibling,
+            'position' => 'before',
+            'expectedContent' => [$bogusOne, $oItem, $oSibling, $bogusTwo],
+            ];
+        yield 'insert after' => [
+            'initialContent' => [$bogusOne, $oSibling, $bogusTwo],
+            'oItem' => $oItem,
+            'oSibling' => $oSibling,
+            'position' => 'after',
+            'expectedContent' => [$bogusOne, $oSibling, $oItem, $bogusTwo],
+            ];
+        yield 'append if not found' => [
+            'initialContent' => [$bogusOne, $oSibling, $bogusTwo],
+            'oItem' => $oItem,
+            'oSibling' => $oOrphan,
+            'position' => 'before',
+            'expectedContent' => [$bogusOne, $oSibling, $bogusTwo, $oItem],
+            ];
+    }
+
+    /**
+     * @test
+     *
+     * @param array $contents
```

Comment:
For the cases where we already have a (working) native type declaration and where the PHPDoc comment does not add anything substantial over the native annotation, please drop the now-redundant PHPDoc annotation.